smartsim/_core/entrypoints/file_operations.py

```diff
@@ -0,0 +1,272 @@
+
+        source path: The tagged files the search and replace operations to be
+            performed upon
+        dest path: Optional destination for configured files to be written to
+        tag_delimiter: tag for the configure operation to search for, defaults to
+            semi-colon e.g. ";"
+        param_dict: A dict of parameter names and values set for the file
+
+    """
+
+    _check_path(parsed_args.source)
+    if parsed_args.dest:
+        _check_path(parsed_args.dest)
+
+    tag_delimiter = ";"
+    if parsed_args.tag_delimiter:
+        tag_delimiter = parsed_args.tag_delimiter
+
+    decoded_dict = base64.b64decode(literal_eval(parsed_args.param_dict))
+    param_dict = pickle.loads(decoded_dict)
+
+    if not param_dict:
+        raise ValueError("param dictionary is empty")
+    if not isinstance(param_dict, dict):
+        raise TypeError("param dict is not a valid dictionary")
+
+    def _get_prev_value(tagged_line: str) -> str:
+        split_tag = tagged_line.split(tag_delimiter)
+        return split_tag[1]
+
+    def _is_ensemble_spec(tagged_line: str, application_params: dict[str, str]) -> bool:
+        split_tag = tagged_line.split(tag_delimiter)
+        prev_val = split_tag[1]
+        if prev_val in application_params.keys():
+            return True
+        return False
+
+    edited = []
+    used_params = {}
+
+    # Set the tag for the application writer to search for within
+    # tagged files attached to an entity.
+    regex = "".join(("(", tag_delimiter, ".+", tag_delimiter, ")"))
+
+    # Set the lines to iterate over
+    with open(parsed_args.source, "r+", encoding="utf-8") as file_stream:
+        lines = file_stream.readlines()
+
+    unused_tags = collections.defaultdict(list)
+
+    # Replace the tagged parameters within the file attached to this
+    # application. The tag defaults to ";"
+    for i, line in enumerate(lines, 1):
+        while search := re.search(regex, line):
+            tagged_line = search.group(0)
+            previous_value = _get_prev_value(tagged_line)
+            if _is_ensemble_spec(tagged_line, param_dict):
+                new_val = str(param_dict[previous_value])
+                line = re.sub(regex, new_val, line, 1)
+                used_params[previous_value] = new_val
+
+            # if a tag_delimiter is found but is not in this application's
+            # configurations put in placeholder value
+            else:
+                tag_delimiter_ = tagged_line.split(tag_delimiter)[1]
+                unused_tags[tag_delimiter_].append(i)
+                line = re.sub(regex, previous_value, line)
+                break
+        edited.append(line)
+
+    lines = edited
+
+    # write configured file to destination specified. Default is an overwrite
+    if parsed_args.dest:
+        file_stream = parsed_args.dest
+
+    with open(parsed_args.source, "w+", encoding="utf-8") as file_stream:
```

Comment:
in the original code, there wasn't a destination specified, it would just overwrite the files. I will change it so that it always writes it to the destination